dynamic_library/CMakeLists.txt

```diff
@@ -3,9 +3,12 @@ leatherman_dependency(util)
 if(WIN32)
     leatherman_dependency(windows)
 else()
-  if(NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+  if(NOT CMAKE_SYSTEM_NAME MATCHES "FreeBSD|OpenBSD")
       add_leatherman_deps(dl)
   endif()
+  if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
```

Comment:
OK, it looks like we only add -fPIC with GCC, not with Clang. You can probably just move this entire "if FreeBSD" block into the Clang block in that file.